client/client-lib/src/api.rs

```diff
@@ -547,16 +547,20 @@ impl<C: JsonRpcClient> WsFederationApi<C> {
             futures.push(member.request(method, &params));
         }
 
-        // Delegates the response handling to the `QueryStrategy` which can
+        // Delegates the response handling to the `QueryStrategy` with an exponential back-off
+        // with every new set of requests
+        let mut delay_ms = 10;
         loop {
             match futures.next().await {
                 Some(result) => match strategy.process(result) {
-                    QueryStep::Request(peers) => {
+                    QueryStep::Retry(peers) => {
                         for member in &self.members {
                             if peers.contains(&member.peer_id) {
                                 futures.push(member.request(method, &params));
                             }
                         }
+                        sleep(Duration::from_millis(delay_ms)).await;
```

Comment:
Are we wasting 10ms here the first time around?

Reply:
Didn't find anything else, it's only a small nit, merging.

Reply:
Nope, it's possible the function succeeds without sleeping.